src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -48,12 +49,60 @@ export class RuntimeLanguageContext {
     }
 
     public convertLanguage(languageId?: string) {
-        languageId = languageId === CodeWhispererConstants.typescript ? CodeWhispererConstants.javascript : languageId
-        if (!languageId) {
+        let mappedId: string | undefined
+        switch (languageId) {
+            case CodeWhispererConstants.typescript:
+                mappedId = CodeWhispererConstants.javascript
+                break
+            case 'javascriptreact':
+                mappedId = CodeWhispererConstants.jsx
+                break
+            default:
+                mappedId = languageId
+                break
+        }
+
+        if (!mappedId) {
             return 'plaintext'
         }
 
-        return languageId
+        return mappedId
+    }
+
+    /**
+     * This method should be called right before calling cwspr API to map some language dialet
+     * e.g. jsx(javascriptreact), typescript into javascript
+     * while keeping the source language name (jsx, typescript here) in client side as we send telemetry metrics with its source language name
+     */
+    public covertCwsprRequest<
```

Comment:
Could you rename this function?

Reply:
ack

Reply:
might need brainstorming on this
is mapRequestToCodeWhispererConsumable Ok?

Reply:
how about `mapRequestLanguageToParentLanguage`

Reply:
renamed to `mapToRuntimeLanguage`